packages/bot-web-ui/src/stores/blockly-store.ts

```diff
@@ -2,41 +2,68 @@ import { action, observable } from 'mobx';
 import { onWorkspaceResize } from '@deriv/bot-skeleton';
 import { tabs_title } from 'Constants/bot-contents';
 import { storeSetting, getSetting } from 'Utils/settings';
+import RootStore from './root-store';
 
-export default class MainContentStore {
-    constructor(root_store) {
+export interface IBlocklyStore {
+    is_loading: boolean;
+    active_tab: string;
+    setActiveTab: (tab: string) => void;
+    setContainerSize: () => void;
+    onMount: () => void;
+    onUnmount: () => void;
+    startLoading: () => void;
+    getCachedActiveTab: () => void;
+    endLoading: () => void;
+}
+
+export default class BlocklyStore implements IBlocklyStore {
+    root_store: RootStore;
+
+    constructor(root_store: RootStore) {
         this.root_store = root_store;
     }
 
+    @observable is_loading = false;
+
     @observable active_tab = tabs_title.WORKSPACE;
 
     @action.bound
-    setActiveTab(tab) {
+    setActiveTab(tab: string): void {
         this.active_tab = tab;
         storeSetting('active_tab', this.active_tab);
     }
 
     @action.bound
-    setContainerSize() {
+    setContainerSize(): void {
         if (this.active_tab === tabs_title.WORKSPACE) {
             onWorkspaceResize();
         }
     }
 
     @action.bound
-    onMount() {
+    onMount(): void {
         window.addEventListener('resize', this.setContainerSize);
     }
 
     @action.bound
-    getCachedActiveTab() {
+    getCachedActiveTab(): void {
         if (getSetting('active_tab')) {
             this.active_tab = getSetting('active_tab');
         }
     }
 
     @action.bound
-    onUnmount() {
+    onUnmount(): void {
         window.removeEventListener('resize', this.setContainerSize);
     }
+
+    @action.bound
+    startLoading(): void {
```

Comment:
Can't we have a `setLoading` action that accept status and set the status based on the args instead of having two separate action for start and end the loading?

Reply:
added a todo, will fix it when we refactor trade engine